runtime/convert_test.go

```diff
@@ -1,71 +1,59 @@
 package runtime_test
 
 import (
-	"encoding/json"
-	"fmt"
-	"reflect"
 	"testing"
 
-	"github.com/golang/protobuf/proto"
-	"github.com/golang/protobuf/ptypes/duration"
-	"github.com/golang/protobuf/ptypes/timestamp"
+	durationpb "github.com/golang/protobuf/ptypes/duration"
+	timestamppb "github.com/golang/protobuf/ptypes/timestamp"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"google.golang.org/protobuf/proto"
 )
 
 func TestConvertTimestamp(t *testing.T) {
 	specs := []struct {
 		name    string
 		input   string
-		output  *timestamp.Timestamp
-		wanterr error
+		output  *timestamppb.Timestamp
+		wanterr bool
 	}{
 		{
 			name:  "a valid RFC3339 timestamp",
 			input: `"2016-05-10T10:19:13.123Z"`,
-			output: &timestamp.Timestamp{
+			output: &timestamppb.Timestamp{
 				Seconds: 1462875553,
 				Nanos:   123000000,
 			},
-			wanterr: nil,
+			wanterr: false,
 		},
 		{
 			name:    "invalid timestamp",
 			input:   `"05-10-2016T10:19:13.123Z"`,
 			output:  nil,
-			wanterr: fmt.Errorf(`bad Timestamp: parsing time "05-10-2016T10:19:13.123Z" as "2006-01-02T15:04:05.999999999Z07:00": cannot parse "0-2016T10:19:13.123Z" as "2006"`),
+			wanterr: true,
 		},
 		{
-			name:   "JSON number",
-			input:  "123",
-			output: nil,
-			wanterr: &json.UnmarshalTypeError{
-				Value:  "number",
-				Type:   reflect.TypeOf("123"),
-				Offset: 3,
-			},
+			name:    "JSON number",
+			input:   "123",
+			output:  nil,
+			wanterr: true,
 		},
 		{
-			name:   "JSON bool",
-			input:  "true",
-			output: nil,
-			wanterr: &json.UnmarshalTypeError{
-				Value:  "bool",
-				Type:   reflect.TypeOf("123"),
-				Offset: 4,
-			},
+			name:    "JSON bool",
+			input:   "true",
+			output:  nil,
+			wanterr: true,
 		},
 	}
 
 	for _, spec := range specs {
 		t.Run(spec.name, func(t *testing.T) {
 			ts, err := runtime.Timestamp(spec.input)
-			if spec.wanterr != nil {
-				if !reflect.DeepEqual(err, spec.wanterr) {
-					t.Errorf("got unexpected error\n%#v\nexpected\n%#v", err, spec.wanterr)
-				}
-				return
-			}
-			if !proto.Equal(ts, spec.output) {
+			switch {
+			case err != nil && !spec.wanterr:
+				t.Errorf("got unexpected error\n%#v", err)
+			case err == nil && spec.wanterr:
+				t.Errorf("did not error when expecte")
+			case !proto.Equal(ts, spec.output):
```

Comment:
Replaced direct error comparison with boolean test (we don't care what the error looks like).